src/main/java/org/humancellatlas/ingest/biomaterial/web/BiomaterialController.java

```diff
@@ -57,4 +64,46 @@
     PersistentEntityResource resource = assembler.toFullResource(entity);
     return ResponseEntity.accepted().body(resource);
   }
+
+  @RequestMapping(path = "biomaterials/{id}/inputBiomaterial/{input_id}", method = RequestMethod.PUT)
+  ResponseEntity<Resource<?>> addInputBiomaterial(@PathVariable("id") Biomaterial biomaterial,
+                                                        @PathVariable("input_id") Biomaterial inputBiomaterial,
+                                                        @RequestBody Process process,
+                                                        PersistentEntityResourceAssembler assembler) {
+    biomaterialService.addInputBiomaterial(inputBiomaterial, process, biomaterial);
+    PersistentEntityResource resource = assembler.toFullResource(biomaterial);
+    return ResponseEntity.accepted().body(resource);
+
+  }
+
+  @RequestMapping(path = "biomaterials/{id}/inputBiomaterials", method = RequestMethod.GET)
+  ResponseEntity<PagedResources> getInputBiomaterials(@PathVariable("id") Biomaterial biomaterial,
+                                                   Pageable pageable,
+                                                   final PersistentEntityResourceAssembler resourceAssembler) {
+    Page<Biomaterial> inputBiomaterials = biomaterialService.getInputBiomaterials(biomaterial, pageable);
+    return ResponseEntity.ok(pagedResourcesAssembler.toResource(inputBiomaterials, resourceAssembler));
+  }
+
+  @RequestMapping(path = "biomaterials/{id}/inputBiomaterials/{input_id}", method = RequestMethod.DELETE)
+  ResponseEntity<Resource<?>> deleteInputBiomaterial(@PathVariable("id") Biomaterial biomaterial,
+                                                  @PathVariable("input_id") Biomaterial inputBiomaterial,
+                                                  PersistentEntityResourceAssembler assembler) {
+    biomaterialService.deleteInputBiomaterial(biomaterial, inputBiomaterial);
+    return ResponseEntity.accepted().build();
+
+  }
+
+  @PostMapping(path = "/biomaterials/query")
```

Comment:
POST was used here because it's not possible to send a body thru Angular's HTTPClient. We could convert everything to be GET parameters but I think it's easier and cleaner to have those in JSON Body. Let me know what you think